plugins/chat/app/jobs/regular/chat/notify_mentioned.rb

```diff
@@ -145,13 +145,43 @@ def process_mentions(user_ids, mention_type)
         memberships = get_memberships(user_ids)
 
         memberships.each do |membership|
-          mention = ::Chat::Mention.find_by(user: membership.user, chat_message: @chat_message)
+          mention = find_mention(@chat_message, mention_type, membership.user.id)
           if mention.present?
             create_notification!(membership, mention, mention_type)
             send_notifications(membership, mention_type)
           end
         end
       end
+
+      def find_mention(chat_message, mention_type, user_id)
+        mention_klass = resolve_mention_klass(mention_type)
+
+        target_id = nil
+        if mention_klass == ::Chat::UserMention
+          target_id = user_id
+        elsif mention_klass == ::Chat::GroupMention
+          begin
+            target_id = Group.where("LOWER(name) = ?", "#{mention_type}").first.id
```

Comment:
Nitpick: if you only care about some fields about 1 record, you can use `#pick` and list the fields you want.

```suggestion
            target_id = Group.where("LOWER(name) = ?", "#{mention_type}").pick(:id)
```

Reply:
Sorry, merged the PR before I saw this. I've fixed it in a follow-up – https://github.com/discourse/discourse/pull/25301.